libraries/lib-module-manager/AsyncPluginValidator.h

```diff
@@ -0,0 +1,64 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file AsyncPluginValidator.h
+
+  @author Vitaly Sverchinsky
+
+  Part of lib-module-manager library
+
+**********************************************************************/
+
+#pragma once
+
+#include <wx/string.h>
+#include <memory>
+
+class PluginDescriptor;
+
+
+/**
+ * \brief Used to perform plugin validation in a dedicated process.
+ * Once instantiated, client should call AsyncPluginValidation::Validate
+ * for plugin that needs to be validated. When done,
+ * AsyncPluginValidation will notify caller via Delegate.
+ */
+class MODULE_MANAGER_API AsyncPluginValidator final
+{
+   class Impl;
+   std::shared_ptr<Impl> mImpl;
+public:
+   /**
+    * \brief Used to talk back to calling side
+    */
```

Comment:
Comments should explain that the callbacks occur on the main thread when the event loop idles or is yielded to.